src/dasm.lua

```diff
@@ -106,7 +106,10 @@ local int_ct = ffi.typeof'int'
 local function convert_arg(arg)   --dasm_put() accepts only int32 varargs.
 	if type(arg) == "number" then  --but we make it accept uint32 too by normalizing the arg.
 		arg = bit.tobit(arg)        --non-number args are converted to int32 according to ffi rules.
-	end
+        elseif type(arg) == "cdata" and ffi.sizeof(arg) > 4 then
```

Comment:
Ah, nice hack.  Catches struct references and int64 values and such.  12/10 good patch would apply